lib/chef/application/client.rb

```diff
@@ -105,10 +105,12 @@ class Chef::Application::Client < Chef::Application
 
   unless Chef::Platform.windows?
     option :daemonize,
-      :short => "-d",
-      :long => "--daemonize",
-      :description => "Daemonize the process",
-      :proc => lambda { |p| true }
+      :short => "-d [WAIT]",
+      :long => "--daemonize [WAIT]",
+      :description =>
+        "Daemonize the process. Accepts an optional integer which is the " \
+        "number of seconds to wait before the first daemonized run.",
+      :proc => lambda {|wait| wait =~ /^\d+$/ ? wait.to_i : true }
```

Comment:
It doesn't feel weird to me (lots of programs have command line options with optional arguments), but I don't have a strong opinion if others think it should be a separate option. @tyler-ball and I tried `--daemonize-run-immediate` before settling on this implementation.